pkg/keystone/keystone.go

```diff
@@ -433,6 +461,9 @@ func (d *keystone) guessScope(ba *gophercloud.AuthOptions) AuthenticationError {
 // It returns the authorization context
 func (d *keystone) authenticate(authOpts gophercloud.AuthOptions, asServiceUser bool, rescope bool) (*policy.Context, string, AuthenticationError) {
 	// check cache, which does not work if tokens are rescoped
+	if authOpts.ApplicationCredentialName != "" || authOpts.ApplicationCredentialID != "" {
```

Comment:
The reason for this is because the token request gets called by default as a service user.  The default check for this, as a service user, is to grab the authOpts.scope, which will cause an error when using application credentials.  I couldn't see a way to have the token request called without asServiceUser set to true.  Open to suggestions on this one.